parboiled-core/src/main/java/org/parboiled/common/Utils.java

```diff
@@ -35,6 +35,8 @@
     public static final Float[] EMPTY_FLOAT_OBJECT_ARRAY = new Float[0];
     public static final Double[] EMPTY_DOUBLE_OBJECT_ARRAY = new Double[0];
     public static final Boolean[] EMPTY_BOOLEAN_OBJECT_ARRAY = new Boolean[0];
+    public static HashMap<Class<?>, Class<?>> PRIMITIVE_TO_BOXED;
```

Comment:
I don't think we really need the HashMap.
A simple `if ... else` chain would do the same, not require more lines of code, probably be even slightly faster and automatically thread-safe.
Could you give it a try?